### PR TITLE
Add support for UUID type in PostgreSQL connector

### LINKED ISSUE #647: Support UUID in PostgreSQL connector
UUID columns in Postgresql tables are silently dropped in Presto. This was previously reported in  prestodb/presto#10951 and @findepi gave implementation directions.

I will probably submit a PR for this in the near-term since this is a feature that I need. I'd like to gather some opinions though on either if UUID types should be represented as a binary(16) or a char(36).

### PR DESCRIPTION
**Pull request by pgagnon**
Closes #647

### CODE REVIEW

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 1]
```diff
@@ -69,10 +69,12 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
 import java.util.function.BiFunction;
 
 import static com.fasterxml.jackson.core.JsonFactory.Feature.CANONICALIZE_FIELD_NAMES;
 import static com.fasterxml.jackson.databind.SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS;
+import static io.airlift.slice.Slices.wrappedLongArray;
 import static io.airlift.slice.Slices.utf8Slice;
```
Comment (ebyhr) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: Wrong order.

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 2]
```diff
@@ -654,6 +656,21 @@ private DataTypeTest jsonTestCases(DataType<String> jsonDataType)
...
+    }
+
+    private DataTypeTest uuidTestCases(DataType<java.util.UUID> uuidDataType)
+    {
+        return DataTypeTest.create()
+                .addRoundTrip(uuidDataType, java.util.UUID.fromString("00000000-0000-0000-0000-000000000000"))
+                .addRoundTrip(uuidDataType, java.util.UUID.fromString("123e4567-e89b-12d3-a456-426655440000"));
+    }
+
+
```
Comment (ebyhr) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java]: Remove redundant empty line.

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 3]
```diff
@@ -417,6 +426,27 @@ private static SliceWriteFunction typedVarcharWriteFunction(String jdbcTypeName)
         };
     }
 
+    private static SliceWriteFunction uuidWriteFunction()
+    {
+        return (statement, index, value) -> {
+            UUID uuid = UUID.nameUUIDFromBytes(value.getBytes());
```
Comment (electrum) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: This is not correct. It’s creating a UUID using an MD5 of the bytes. Unfortunately, there is no direct conversion from bytes. You need to convert to longs and use the constructor.

The round trip tests should have caught that the value didn’t round trip properly. It’s concerning that they did not.

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 4]
```diff
@@ -417,6 +427,29 @@ private static SliceWriteFunction typedVarcharWriteFunction(String jdbcTypeName)
         };
     }
 
+    private static SliceWriteFunction uuidWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long mostSig = value.getLong(0);
```
Comment (electrum) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: Nit: let's inline `mostSig` and `leastSig` (like we do in `UuidType` and `UuidOperators`)

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java [hunk 5]
```diff
@@ -417,6 +427,29 @@ private static SliceWriteFunction typedVarcharWriteFunction(String jdbcTypeName)
...
+            long leastSig = value.getLong(SIZE_OF_LONG);
+            UUID uuid = new UUID(mostSig, leastSig);
+            statement.setObject(index, uuid, java.sql.Types.OTHER);
```
Comment (electrum) [inline presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: Nit: no need for the `java.sql.` prefix on `Types`

File: presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java (hunk 3 above)
Comment (pgagnon) [reply presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java]: Sorry... the test should have caught that, but I got lazy and missed the forest for the trees.

This part used to fail with a JVM crash, and when I got it fixed I figured that a bout of manual testing should do and then let travis handle the rest.

Evidently that was premature on my part. 😓

**Comment by pgagnon** [comment]
CI has passed. 🎉

**Comment by electrum** [comment]
Merged, thanks!

**Pull request merged**